pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -191,6 +186,15 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// fetch the clusterdeployment and bail out if there's an outgoing migration annotation again
```

Comment:
I gather the idea is to check/bail before any write operation? If so, I don't think this block is necessary, since `ShouldReissue` is read-only, and you're doing another check right before `IssueCertificate`.

Reply:
nice catch! i had it in my head that `IssueCertificate` did more than it does